source/features/forked-to.tsx

```diff
@@ -0,0 +1,154 @@
+import './more-dropdown.css';
+import React from 'dom-chef';
+import select from 'select-dom';
+import features from '../libs/features';
+import cache from '../libs/cache';
+import {getRepoURL, getUsername} from '../libs/utils';
+import {isOwnRepo} from '../libs/page-detect';
+
+const currentRepo = getRepoURL();
+
+async function init(): Promise<void> {
+	onForkDialogOpened();
+
+	if (isOwnRepo()) {
+		onForkedPage();
+	}
+
+	await checkForks();
+}
+
+// Check for cached forks.
+async function checkForks(): Promise<void> {
+	const repo = getOriginalRepo();
+	const cached = await getCache(repo);
+	const validForks = cached.filter(validateFork);
+	for (const fork of await Promise.all(validForks)) {
+		if (fork !== currentRepo) {
+			appendHtml(fork);
+			storeCache(repo, fork);
```

Comment:
It's a key per original repo. The forks are in array as value.

Reply:
Indeed, but each loop is getting the whole array, adding one key and saving it. 

And: you’re not awaiting storeCache so all of these requests might happen not in the order you want.

And: I don’t think invalidated forks are removed from the cache.

I think it’s best to validate the whole cache and just store the whole array as-is, instead of merging it with the existing one. It’s safer.

Reply:
I'm on it.

Reply:
This wasn’t resolved. You’re still merging the array with the cached one. You probably don’t need the helper function at all, just store the validated cache into the cached

Reply:
Merging the array after filtering is done here: https://github.com/sindresorhus/refined-github/pull/2141/files#diff-ff16301f232100bd87e910eb98614f66R78